src/com/goide/highlighting/GoAnnotator.java

```diff
@@ -193,6 +200,23 @@ else if (element instanceof GoFunctionDeclaration) {
         }
       }
     }
+    else if (element instanceof GoIndexOrSliceExpr) {
+      GoIndexOrSliceExpr slice = (GoIndexOrSliceExpr)element;
+      GoExpression expr = slice.getExpression();
+      if (expr == null || slice.getIndices().third == null) {
+        return;
+      }
+      if (GoTypeUtil.isString(expr.getGoType(null))) {
+        ASTNode[] colons = slice.getNode().getChildren(TokenSet.create(GoTypes.COLON));
+        if (colons.length == 2 && slice.getRbrack() != null) {
+          PsiElement secondColon = colons[1].getPsi();
+          Annotation annotation =
+            holder.createErrorAnnotation(new TextRange(secondColon.getTextOffset(), slice.getRbrack().getTextOffset()),
+                                         "Slice of string c annotation have 3-index");
```

Comment:
"c"? "annotation"?
It's easy to think of error messages in annotator – just repeat the compiler:

`"Invalid operation " + expr.getText() + " (3-index slice of string)"`